src/java/voldemort/server/StoreRepository.java

```diff
@@ -185,7 +185,7 @@ public void addStorageEngine(StorageEngine<ByteArray, byte[], byte[]> engine) {
         // register streaming stats object for the store
         if(streamingStatsMap != null) {
             JmxUtils.unregisterMbean(JmxUtils.createObjectName(this.getClass().getCanonicalName(),
-                                                               storeName));
+                                                               storeName + "-streaming-stats"));
```

Comment:
Greg, I really appreciate you making this change. 

I would prefer to create a method for this, I will do this in a later commit.